version: '3.1'

services:
  db:
    restart: always
    networks:
      - internal
      - web

  nginx:
    image: nginx
    container_name: achacasa_nginx
    restart: always
    volumes:
      - ./config:/etc/nginx/conf.d
      - staticfiles:/static
    networks:
      - internal
      - web
    depends_on:
      - web
    labels:
      - 'traefik.http.routers.achacasa.rule=Host(`achacasa.online`)'
      - 'traefik.http.routers.achacasa.entrypoints=web'
      - 'traefik.http.routers.achacasa_secure.rule=Host(`achacasa.online`)'
      - 'traefik.http.routers.achacasa_secure.tls.certresolver=achacasa'
      - 'traefik.http.routers.achacasa_secure.entrypoints=websecure'
      - 'traefik.enable=true'
      - 'traefik.docker.network=web'

  web:
    build: .
    container_name: achacasa_web
    restart: always
    command: python manage.py runserver 0.0.0.0:8000
    networks:
      - internal
    volumes:
      - staticfiles:/code/static
    depends_on:
      - db
    expose:
      - '8000'

volumes:
  staticfiles:

networks:
  web:
    external: true
  internal:
    external: false
